metrics/README.md

```diff
@@ -0,0 +1,203 @@
+CI will either not spot regressions that are below that noise factor, or will cause
+false failures, which are very undesirable in a CI.
+
+### `master` branch trending
+
+The master branch tracking CI (which tracks and collects data for each merge into
+the master branch) collects data for:
+- future historical reference
+- longer term trend tracking
+- more complex or longer running test cases
+- more complex regression checking
+
+As the master branch does not have the same urgency to provide feedback to a pending PR,
+it can take the luxury of running tests for longer, or running more complex tests.
+
+### Developer pre-checking
+
+The PR regression check scripts can be executed 'by hand', and thus are availalble
+for developers to use as a 'pre-check' before submitting a PR. It might be prudent for
+developers to follow this procedure particularly for large architectural or version changes
+of components.
+
+## Stability or Performance?
+
+When forming, or configuring, metrics tests, often we have to make a choice or compromise
+on if we want the test to take a more repeatable measurement (less noise and variance), or if
+we want to measure the 'best performance'.
+
+Generally for CI regression checking, we prefer stability over performance, as that allows
+us a more accurate (narrow bound) check for regressions.
+
+> **NOTE** It should thus be noted that if you are gathering data to discuss performance,
+> then you may *not* want to use the CI metrics data, as this may be favoring stable results
+> over best performance.
+
+## Requirements
+
+To try and maintain the quality of the metrics data gathered and the accuracy of the CI
+regression checking, we try to define and stick to some 'quality measures' for our metrics.
+
+### For PR checks
+
+The PR CI is generally required to execute within a 'reasonable time' to provide timely
+feedback to the developers (and not stall the review and development process). To that
+end, we relax the quality requirements of the PR CI. The quality requirements are:
+- <= 5% run to run variance
+- <= 5 minutes runtime per test
+
+### For master tracking
+
+For master branch CI tracking, we have the luxury that there is no strict time requirement,
+therefore we can use stricter requirements for the data generation, which brings us some
+benefits:
+- With finer data we can do some finer regression checks
+- With access to more historical data, we can add trend checks
+    - such as catching 'death by a thousand cuts', such as a 1% drop per PR or similar
+- Allows us to run tests that would inherently take longer than a CI PR metric is allowed
+
+To that end, we set the quality requiments tighter than the PR CI, but allow more runtime:
+- <= 2% run to run variance
+- <= 20 minutes runtime per test
+
+## Categories
+
+Kata Container metrics tend to fall into a set of categories, and we organise the tests
+wtihin this folder as such.
+
+Each sub-folder containers its own `README` detailing its own tests.
```

Comment:
typo: s/containers/contains?

Reply:
heh - I make that typo quite often - too many containers in my brain....